tests/lib/nested.sh

```diff
@@ -64,18 +89,91 @@ create_nested_core_vm(){
     fi
 }
 
-start_nested_vm(){
-    systemd_create_and_start_unit nested-vm "${QEMU} -m 2048 -nographic -net nic,model=virtio -net user,hostfwd=tcp::8022-:22 -drive file=/tmp/work-dir/ubuntu-core.img,if=virtio,cache=none,format=raw -drive file=${PWD}/assertions.disk,if=virtio,cache=none,format=raw"
+start_nested_core_vm(){
+    systemd_create_and_start_unit nested-vm "${QEMU} -m 2048 -nographic \
+        -net nic,model=virtio -net user,hostfwd=tcp::$SSH_PORT-:22 \
+        -drive file=$WORK_DIR/ubuntu-core.img,if=virtio,cache=none,format=raw \
+        -drive file=${PWD}/assertions.disk,if=virtio,cache=none,format=raw \
+        -monitor tcp:127.0.0.1:$MON_PORT,server,nowait -usb"
     if ! wait_for_ssh; then
         systemctl restart nested-vm
     fi
 }
 
-destroy_nested_core_vm(){
+create_nested_classic_vm(){
+    mkdir -p "$WORK_DIR"
+
+    # Get the cloud image
+    IMAGE_URL=$(get_image_url_for_nested_vm)
+    wget -P "$WORK_DIR" "$IMAGE_URL"
+    IMAGE=$(ls $WORK_DIR/*.img)
```

Comment:
Suprising shellcheck did not complain here. I think you need to make sure there's only one image, this should be sufficient:
```
test "$(echo "$IMAGE" | wc -l)" = "1"
```